scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -49,52 +65,141 @@ namespace <%= namespace %> {
+            MutationQuery query = new MutationQuery();
+
+            List<CheckoutLineItemInput> newLineItemInput = CartLineItems.ConvertToCheckoutLineItemInput(LineItems.New());
 
-            url.Append("http://");
-            url.Append(Client.Domain);
-            url.Append("/cart/");
+            DefaultQueries.checkout.Create(query, newLineItemInput);
 
-            foreach(CartLineItem lineItem in LineItems.All()) {
-                if (hasLineItem) {
-                    url.Append(",");
+            Client.Mutation(query, (Mutation response, List<string> errors, string httpError) => {
+                if (errors != null || httpError != null) {
+                    callback(errors, httpError);
+                } else {
+                    LineItems.ClearNew();
+
+                    UpdateCheckout(response.checkoutCreate().checkout());
+                    
+                    CheckoutPoll(callback);
                 }
+            });
+        }
 
-                hasLineItem = true;
+        private void CheckoutUpdate(Action<List<string>, string> callback) {
+            MutationQuery query = new MutationQuery();
+            bool doAdd = LineItems.New().Count > 0;
+            bool doModify = LineItems.Modified().Count > 0;
+            bool doDelete = LineItems.Deleted().Count > 0;
 
-                string variantId = System.Text.Encoding.UTF8.GetString(
-                    Convert.FromBase64String(lineItem.VariantId)
-                );
+            if (doAdd) {
+                DefaultQueries.checkout.LineItemsAdd(query, CurrentCheckout.id(), CartLineItems.ConvertToCheckoutLineItemInput(LineItems.New()));
+            }
 
-                // convert from variant gid to variant id
-                string[] variandIdSplit = variantId.Split('/');
+            if (doModify) {
+                DefaultQueries.checkout.LineItemsUpdate(query, CurrentCheckout.id(), CartLineItems.ConvertToCheckoutLineItemUpdateInput(LineItems.Modified()));
+            }
 
-                url.Append(variandIdSplit[variandIdSplit.Length - 1]);
-                url.Append(":");
-                url.Append(lineItem.Quantity);
+            if (doDelete) {
+                DefaultQueries.checkout.LineItemsRemove(query, CurrentCheckout.id(), CartLineItems.ConvertToCheckoutIds(LineItems.Deleted()));
             }
 
-            url.Append("?access_token=");
-            url.Append(Client.AccessToken);
+            Client.Mutation(query, (Mutation response, List<string> errors, string httpError) => {
+                if (errors != null || httpError != null) {
+                    callback(errors, httpError);
+                } else {
+                    LineItems.ClearNew();
+                    LineItems.ClearModified();
+                    LineItems.ClearDeleted();
+
+                    if (doAdd) {
+                        UpdateCheckout(response.checkoutLineItemsAdd().checkout());
+                    } else if (doModify) {
```

Comment:
Yes there is a case like that but in both cases `Checkout` will return the same thing so there's no need to do it twice.